openforcefield/tests/test_forcefield.py

```diff
@@ -1096,8 +1096,38 @@ def test_nonbonded_method_resolution(self,
                 forcefield.get_parameter_handler('Electrostatics', {}).method = electrostatics_method
                 omm_system = forcefield.create_openmm_system(topology)
 
+    def test_parameter_handler_lookup(self):
+        """Ensure __getitem__ lookups work"""
+        forcefield = ForceField('test_forcefields/smirnoff99Frosst.offxml')
+
+        handlers_before = sorted(forcefield._parameter_handlers)
+
+        for val in handlers_before:
+            looked_up_handler = forcefield[val]
+            assert isinstance(looked_up_handler, ParameterHandler)
+
+        handlers_after = sorted(forcefield._parameter_handlers)
 
+        assert handlers_before == handlers_after
 
+    def test_unregistered_parameter_handler_lookup(self):
+        """Ensure __getitem__ lookups do not register new handlers"""
+        forcefield = ForceField('test_forcefields/smirnoff99Frosst.offxml')
+        unregistered_handler = 'LibraryCharges'
+
+        assert unregistered_handler not in forcefield._parameter_handlers
+        forcefield[unregistered_handler]
```

Comment:
Do we want this silently returning `None`, or should it throw something like a `KeyError`?

Reply:
Good catch - tried to address it in ab2587a